### DISCUSSION TITLE
how to add tls cert for pgbouncer

### DISCUSSION THREAD

**Discussion by riker1** [autobase-tech/autobase#685] [Q&A] [answered]
I want to use SSL for my pgbouncer connection and I see that I've got a line for pgbouncer_client_tls_key_file and pgbouncer_client_tls_cert_file. 

 I am not using consul but I noticed that there's a ref to `You can enable TLS encryption by dropping a CA certificate, server certificate, and server key in roles/consul/files/`
 
 should I create a files folder in the pgbouncer role? if so would I reference it as {{ roles/pgbouncer/files/cert.crt etc?
 
 thanks in advance --est2

**Comment by riker1** [comment, marked as answer]
I think I answered my own question :sweat_smile:

I see under the system variables that I can reference my certs which will then copy them to all nodes. Then I can reference them in the sections requiring TLS certs.

now if I could only figure out why I cannot get pgbackrest to work with my cluster... i'd be home free! :blush:

**Reply by vitabaks** [reply]
Hi @riker1 ! 

Yes, Just upload any files to the [files](https://github.com/vitabaks/postgresql_cluster/tree/1.10.0/files) directory (including certificates) and then define them in the [copy_files_to_all_server](https://github.com/vitabaks/postgresql_cluster/blob/1.10.0/vars/system.yml#L186) variable so that they are copied to the right place and with the right permissions.

**Reply by vitabaks** [reply]
> now if I could only figure out why I cannot get pgbackrest to work with my cluster

Please open a new discussion or issue on this problem and describe the details, I will be happy to help.

**Reply by riker1** [reply]
ok, i just need to get my thoughts together!
